### ISSUE TITLE
Calendar control with Headers as Buttons - Improper name for Header Button as it does not convey what the control does

### ISSUE THREAD

**Issue by meghanking** [microsoft/fluentui#13831]
### Environment Information

- **Package version(s)**: 7.121.4 and earlier
- **Browser and OS versions**: Chrome/Edge and Windows 10. Chrome and Android

### Describe the issue:

When focused on the "2020" Header Button, the accessible name does not convey what the control's purpose. Customer will not know what invoking this button will do or change.

We do not have control of changing or augmenting the current accessible name for this control.

Customer impact as documented by trusted tester: "If the name of the interactive element is not properly announced for screen reader users then they will not understand the functionality of the control and even how to interact with it."

### Please provide a reproduction of the issue in a codepen:

Permalink - https://fabricweb.z5.web.core.windows.net/oufr/7.121.4/#/examples/calendar

- Inline Calendar with week selection and overlaid year picker when month header is clicked
- Inline Calendar with week selection, date boundary (minDate, maxDate), disabled dates (restrictedDates), and overlaid year picker when month header is clicked
- Inline Calendar with month selection and overlaid year picker when month header is clicked
- Inline Calendar with month picker, no day picker, and overlaid year picker when month header is clicked
- Inline Calendar with date boundary (minDate, maxDate), disabled dates (restrictedDates), and overlaid year picker when month header is clicked
- Calendar with workWeekDays = [T W, F, Sa] provided, first day of week = M, and overlaid year picker when month header is clicked
- ...

#### Actual behavior:

Accessible name matches the visible content of "2020"

#### Expected behavior:

In addition to the visible content, the name should convey what action will occur when invoking. In this scenario, the buttons accessible name should convey that this will switch to selecting a year instead of selecting a month

### Documentation describing expected behavior

Note: when picking a new name, make sure to follow the WCAG 2.1 rules for "Label in Name" - https://www.w3.org/WAI/WCAG21/Understanding/label-in-name.html

--

MAS references as supplied by trusted tester:

MAS 1.3.1
https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={54f28d1f-a2d1-4dcd-84e1-5c9b87e8aba4}

MAS 4.1.2
https://microsoft.sharepoint.com/:w:/r/teams/msenable/_layouts/15/WopiFrame.aspx?sourcedoc={248054a6-5e68-4771-9e1e-242fb5025730}

**Comment by paulgildea** [comment]
@lorejoh12 - Can you take look at this?

@meghanking - Are you looking for the ability to provide an accessible name - like an additional API?

**Comment by meghanking** [comment]
@paulgildea I am looking for the default accessible name of these buttons to have wording which also conveys what action the control will perform.

If that is unable to be provided, then I guess having some props where we can pass in a name will suffice. This is necessary as this issue is impacting the accessibility compliance for the products utilizing this control

This would also be true for the similar control that is available within the Date-time Package.
My understanding is our product is currently using Fabric version 6.190.0

**Comment by lorejoh12** [comment]
It looks like this bug is based solely on the office-ui-fabric-react version of Calendar, which is out of date and didn't receive the last round of accessibility fixes.

I would suggest trying the date-time package version.
Here's a deploy link from today that you can run the accessibility pass on: http://fabricweb.z5.web.core.windows.net/pr-deploy-site/refs/pull/14008/merge/date-time/dist/index.html#/examples/calendar

If you turn on Narrator and go to the "2020" button in the month picker, it correctly reads out "2020, select to change the year". If this is not sufficient, please update again with a suggested string that you would prefer and we can take a look at making it more clear

**Comment by meghanking** [comment]
I will let the product team know that the only fix is to update to a the newer date-time package version.

From my short test pass, and from @lorejoh12 's initial comment, that seems like a sufficient accessible name for the "2020" button.

I will bring the other issue I found with the example from the deploy link in another issue item

**Comment by lorejoh12** [comment]
closing for now as by design then! feel free to reopen if there's something new that comes up here.

**Comment by meghanking** [comment]
Makes sense. Thank you for helping us with this issue.
